
# Overview

This repo contains code and data used in the paper "HybridExpress: 
an R/Bioconductor package for comparative transcriptomic analyses of 
hybrids and their progenitors"


Reproducible reports for all the analyses we performed are available as 
a Quarto book at <https://almeidasilvaf.github.io/HybridExpress_paper/>.


## Summary

- Hybridization, the process of crossing individuals from diverse 
genetic backgrounds, plays a pivotal role in evolution, biological 
invasiveness, and crop breeding. 

- At the transcriptional level, hybridization often leads to complex 
non-additive effects, presenting challenges for understanding its 
consequences. Although standard transcriptomic analyses exist to compare 
hybrids to their progenitors, such analyses have not been implemented in a 
software package, hindering reproducibility.

- We introduce *HybridExpress*, an R/Bioconductor package designed to 
facilitate the analysis, visualization, and comparison of gene expression 
patterns in hybrid triplets (hybrids and their progenitors). *HybridExpress* 
provides users with a user-friendly and comprehensive workflow that includes 
all standard comparative analyses steps, including data normalization, 
calculation of midparent expression values, sample clustering, 
expression-based gene classification into categories and classes, 
and overrepresentation analysis for functional terms. 

- We illustrate the utility of *HybridExpress* through comparative 
transcriptomic analyses of cotton allopolyploidization and rice root trait 
heterosis. *HybridExpress* is designed to streamline comparative transcriptomic 
studies of hybrid triplets, advancing our understanding of evolutionary 
dynamics in allopolyploids, and enhancing plant breeding strategies.


**Keywords:** gene expression, transcriptomic shock, allopolyploidy, hybrid vigor, expression-level dominance.

